packages/ui-link/src/Link/style.js

```diff
@@ -0,0 +1,157 @@
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  // get the theme variables object for the component
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+  const { isWithinText, renderIcon, iconPlacement, variant, color } = props
+  const inverseStyle = variant === 'inverse' || color === 'link-inverse'
+  const textDecorationStyle = isWithinText
+    ? componentTheme.textDecorationWithinText
+    : componentTheme.textDecorationOutsideText
+  // If Link is a button or link, it should look clickable
+  const buttonOrLinkStyle =
+    state.elementName === 'button' || state.elementName === 'a'
+      ? {
+          cursor: 'pointer',
+          color: inverseStyle
+            ? componentTheme.colorInverse
+            : componentTheme.color,
+          textDecoration: props.isWithinText
+            ? componentTheme.textDecorationWithinText
+            : componentTheme.textDecorationOutsideText,
+          '&:focus': {
+            color: inverseStyle
+              ? componentTheme.colorInverse
+              : componentTheme.color
+          },
+          '&:hover, &:active': {
+            color: inverseStyle
+              ? componentTheme.colorInverse
+              : componentTheme.hoverColor,
+            textDecoration: props.isWithinText
+              ? componentTheme.hoverTextDecorationWithinText
+              : componentTheme.hoverTextDecorationOutsideText
+          }
+        }
+      : {}
+
+  const inverseStyleLinkSelectors = inverseStyle
+    ? {
+        '&:link, &:visited': {
+          color: componentTheme.colorInverse
+        },
+        '&:link:focus, &:visited:focus': {
+          outlineColor: componentTheme.focusInverseIconOutlineColor
+        },
+        '&:link:hover, &:link:focus, &:link:active, &:visited:hover, &:visited:focus, &:visited:active': {
+          color: componentTheme.colorInverse
+        }
+      }
+    : {}
+
+  const buttonStyle =
+    state.elementName === 'button'
+      ? {
+          appearance: 'none',
+          userSelect: 'text',
+          background: 'none',
+          border: 'none',
+          cursor: 'pointer',
+          fontSize: '1em',
+          margin: 0,
+          padding: 0,
+          textAlign: 'inherit'
+        }
+      : {}
+
+  // If TruncateText is used in Link with icon, align the icon and the text vertically
+  const truncateStyle =
+    props.renderIcon && state.containsTruncateText && state.hasVisibleChildren
+      ? { alignItems: 'center' }
+      : {}
+
+  const linkStyles = {
+    label: 'link',
+    textDecoration: textDecorationStyle,
```

Comment:
I don't think it is needed here: in the css file I can see text-decoration only in case of button or a tags, not by default.

Reply:
good catch :)